pkg/kv/kvserver/replica_application_result.go

```diff
@@ -405,6 +405,8 @@ func (r *Replica) makeReproposal(origP *ProposalData) (reproposal *ProposalData,
 }
 
 func (r *Replica) tryReproposeWithNewLeaseIndex(ctx context.Context, origCmd *replicatedCmd) error {
```

Comment:
Rename with `raftMuLocked` suffix? Let's stick to the naming conventions unless there's a strong reason not to.